infra/elasticsearch/src/client.ts

```diff
@@ -0,0 +1,188 @@
+    /* eslint-disable @typescript-eslint/naming-convention */
+    await client.indices.create({
+      body: {
+        mappings: mappings,
+        settings: {
+          index: {
+            analysis: {
+              analyzer,
+              char_filter,
+              filter,
+              tokenizer,
+            },
+          },
+          number_of_replicas: 1,
+          number_of_shards: 1,
+        },
+      },
+      index: indexName,
+    });
+    logger.info(`Index ${indexName} created.`);
+  } catch (error: unknown) {
+    logger.error("index create", error);
+    throw error;
+  }
+}
+
+export async function version({ client }: UtilsOptions): Promise<void> {
+  const { body } = await client.info<{ version: { number: string } }>();
+  logger.info(body.version.number);
+}
+
+export async function bulkIndexDocuments({
+  client,
+  indexName,
+  documents,
+}: UtilsOptions & { documents: Document[] }): Promise<void> {
+  try {
+    interface BulkApiResponse {
+      errors?: unknown;
+      items: { index: { status: number } }[];
+    }
+
+    const body = documents.flatMap((doc, i) => [
+      {
+        index: {
+          _index: indexName,
+          // if available, use our cdtnId as the actual Elastic document id
+          ...(doc.cdtnId && { _id: doc.cdtnId }),
+          // unless we're in testing mode where we use position
+          ...(process.env.NODE_ENV === "test" && { _id: i }),
+        },
+      },
+      doc,
+    ]);
+
+    const resp = await client.bulk<BulkApiResponse>({
+      body,
+      index: indexName,
+    });
+    if (resp.body.errors) {
+      const errorDocs = resp.body.items.filter(
+        (item) => item.index.status != 201
+      );
+      logger.error(`Errors during indexation : ${JSON.stringify(errorDocs)}`);
+    }
+    logger.info(`Index ${documents.length} documents.`);
+  } catch (error: unknown) {
+    logger.error("index documents", error);
+    throw error;
+  }
+}
+
+export async function indexDocumentsBatched({
+  client,
+  indexName,
+  documents,
+  size = 1000,
+}: UtilsOptions & { documents: Document[]; size: number }): Promise<void> {
+  logger.info(`Loaded ${documents.length} documents`);
+  for (const chunk of chunks(documents, size)) {
+    await bulkIndexDocuments({ client, documents: chunk, indexName });
+  }
+}
+
+export async function deleteOldIndex({
+  client,
+  patterns,
+  timestamp,
+}: UtilsOptions & { patterns: string[]; timestamp: number }): Promise<void> {
+  const { body: indices } = await client.cat.indices<{ index: string }[]>({
+    format: "json",
+  });
+
+  const IndicesToDelete = getIndicesToDelete(patterns, timestamp, indices);
+  const pIndicesToDelete = IndicesToDelete.map(async ({ index }) =>
+    client.indices.delete({ index })
+  );
+
+  return Promise.all(pIndicesToDelete).then(() => {
+    logger.info(`Remove ${pIndicesToDelete.length} old indices`);
+  });
+}
+
+export function* chunks<T = unknown>(
```

Comment:
is ti a best practice to give a Generic an `unknonw` type ?

Reply:
it's more like a temporally workaround seems I do not know what is the type to put heer